lisa/tools/ethtool.py

```diff
@@ -318,7 +336,12 @@ def get_device_channels_info(
         if vcpu_count < device_channel_info.max_channels:
             device_channel_info.max_channels = vcpu_count
 
-        self._device_channel_map[interface] = device_channel_info
+        if interface not in self._device_settings_map.keys():
```

Comment:
The logic is similar, and it can be a method like
```python
def set_device(name: str, device_channels: Optional[DeviceChannel] = None,
        device_features: Optional[DeviceFeatures] = None,
        device_link_settings: Optional[DeviceLinkSettings] = None,
        device_ringbuffer_settings: Optional[DeviceRingBufferSettings] = None,
        device_gro_lro_settings: Optional[DeviceGroLroSettings] = None)
    ...
```

Reply:
Thanks, changed per our offline discussion